app/assets/javascripts/plate-map/plate-layout.js

```diff
@@ -110,6 +110,12 @@ $.widget("DNA.plateLayOut", {
     // object. internally we add to widget.DNA.getPlates.prototype.
     // Helpers are methods which return other methods and objects.
     // add Objects to plateLayOutWidget and it will be added to this object.
+    // set read only well
+    if (this.options.readOnly){
+      this.isReadOnly(true);
+      this.readOnly = true;
```

Comment:
Isn't this redundant?

Reply:
Also, isReadOnly has code for handling false, but never invoked.

Reply:
Also, currently this.readOnly will never be set to false, only the default of null.

Reply:
Actually this part is set up so that when first time putting the plate data into plate map, we can specify readOnly as an input flag without calling $(#...).plateLayOut("isReadOnly", true)

Reply:
If the plate is set up as a read only plate, this.readOnly will be set to True. Else we can also invoke $("#...").plateLayOut("isReadOnly", true/false) to set the this.readOnly to be true or false

Reply:
OK. `this.readOnly = true;` still seems to be redundant though - `this.isReadOnly` should have already set it to true.

Reply:
Yes you are right! I have just removed this.readOnly = true!